### PR TITLE
1163/non intrusive price suggestion

### LINKED ISSUE #1163: Make price estimation non-intrusive
## The problem
There has been negative feedback lately regarding the price estimation feature.

Currently, it replaces the price input with the estimation when it's received from `dex-price-estimator` service.

The summary of raised points are:
1. User input is overwritten without warning
1. Some times the estimation arrives too late, and users end up placing orders with bad prices (might have been solved already)
1. Estimation for larger amounts might return in worse prices, making the `receive amount` smaller than when selling a smaller amount.
1. Users do not know how far off their prices are from the estimation when placing the orders

More details and suggestions can be found on this spreadsheet containing [Mesa feedback and improvement suggestions](https://docs.google.com/spreadsheets/d/1Z8D9Bka4ewojkstrK7bZKu7yB1iof1rnUDxBSQC6cKU/edit#gid=0)

### PR DESCRIPTION
**Pull request by alfetopito**
## Update July 2

- Added swap icon
- No longer showing `infinity` prices
- Added tooltip. Please validate text contents:
screenshot_2020-07-02_13-53-28

## Update Jun 30

- Did a minimal styling, properly aligning things and matching other parts of the app
- Adjusted widget height to fit the new entry
- No longer displaying `infinity` for bad prices. Using `N/A` instead
- Disabling input when `N/A`
- Added base/quote token suffix to be explicit on what this price refers too (although being a bit redundant)
- Added `selling` word to sentence to make it clear what the amount refers to: `Onchain orderbook price for selling <amount> <symbol|name|address>`
- Fixed the displayed symbol in the sentence. Previously it was changing every time the price was inverted
- Displaying second price estimation only when amount is different from default of 1 unit. 

Worst case scenario (long token symbols):
screenshot_2020-06-30_14-49-37

Regular use case (short token symbols):
screenshot_2020-06-30_15-05-39

### TODO:
- [ ] Show % difference between prices (slippage?)
- [x] Add tooltips
- [x] Add swap price buttons

## Proposal

No longer automatically update price inputs with price suggestion.
Instead, show a button and allow the user to make the choice.

Closes #1163

This change adds mostly the logic of fetching the prices and displaying it in a button.

I added 1 price for a single unit and one price for the selected amount.
This way the user has a reference of the `slippage` given his amount.

When switching to the inverse price, the suggestion is also inverted.
When clicking in a button, both prices are filled.

screenshot_2020-06-29_15-09-52

It's hideous, I know. Will work on that.

As always, open for suggestions/feedback.

## TODO:
- [x] Style
- [x] Do not show `infinity` when one of the prices is too big/too small
- [x] More testing, look for edge cases, etc

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr1164--dexreact.review.gnosisdev.com

**Comment by W3stside** [comment]
@alfetopito careful on the height of the div considering the added content aka price suggestion. If it gets too tall (e.g when the user puts in too much sell and gets the over balance warning) the submit button will wrap to the right and look shitty af

File: src/components/TradeWidget/PriceEstimations.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    quoteTokenDecimals,
+  })
+
+  const displayPrice = priceEstimation
+    ? (isPriceInverted ? invertPrice(priceEstimation) : priceEstimation).toFixed(PRICE_ESTIMATION_PRECISION)
+    : '0'
+
+  return (
+    <div
+      style={{ display: 'flex', justifyContent: 'space-between', padding: '0.1em 0 0.1em 0.5em', fontSize: '1.4rem' }}
```
Comment (W3stside) [inline src/components/TradeWidget/PriceEstimations.tsx:77]: You know what to do here

File: src/components/TradeWidget/index.tsx [hunk 2]
```diff
@@ -506,6 +499,12 @@ const TradeWidget: React.FC = () => {
   const defaultValidFrom = trade.validFrom || validFromParam
   const defaultValidUntil = trade.validUntil || validUntilParam
 
+  const [priceShown, setPriceShown] = useState<'INVERSE' | 'DIRECT'>('INVERSE')
+
+  const swapPrices = (): void => {
```
Comment (W3stside) [inline src/components/TradeWidget/index.tsx:504]: Can be one liner lambda

File: src/components/TradeWidget/PriceEstimations.tsx (hunk 1 above)
Comment (alfetopito) [reply src/components/TradeWidget/PriceEstimations.tsx:77]: Yup :)

Silver-tape-quick-hacky-fix-first-revision

**Comment by alfetopito** [comment]
> @alfetopito careful on the height of the div considering the added content aka price suggestion. If it gets too tall (e.g when the user puts in too much sell and gets the over balance warning) the submit button will wrap to the right and look shitty af

Indeed. In that case, what's the best way to handle it? Just increase the default max height?

**Comment by alfetopito** [comment]
> Other than a minor things, like the INFINITE, and probably the tooltip, I think it's good to go.
> 
> The others we can add it to our backlog. Feel free to continue working on them since you are on fire
> But I think we should freeze one version with what we have and deliver

@anxolin addressed your comments. I'll wait for your feedback on the tooltip text before merging, though.
